components/combobox/combobox.jsx

```diff
@@ -970,10 +974,8 @@ class Combobox extends React.Component {
 	handleRequestClose = (event, data) => {
 		if (this.props.events.onRequestClose) {
 			this.props.events.onRequestClose(event, data);
-		}
-
-		if (this.getIsOpen()) {
-			this.setState({ isOpen: false });
+		} else {
```

Comment:
I don't think we should be limiting calling `handleClose` to only when `onRequestClose` is provided.

Instead, does this work?
```
if (this.props.events.onRequestClose) {
	this.props.events.onRequestClose(event, data);
}
if (this.getIsOpen()) {
	this.handleClose(event, { trigger: 'cancel' });
}
```

Reply:
This should also work. I change this accordingly.